packages/core/src/shared/awsClientBuilderV3.ts

```diff
@@ -78,19 +88,37 @@ export class AWSClientBuilderV3 {
         return shim
     }
 
-    public async createAwsService<C extends AwsClient>(
-        type: AwsClientConstructor<C>,
-        options?: Partial<AwsClientOptions>,
-        region?: string,
-        userAgent: boolean = true,
-        keepAlive: boolean = true,
-        settings?: DevSettings
-    ): Promise<C> {
+    private keyAwsService<C extends AwsClient>(serviceOptions: AwsServiceOptions<C>): string {
+        // Serializing certain objects in the args allows us to detect when nested objects change (ex. new retry strategy, endpoints)
+        return [
+            String(serviceOptions.serviceClient),
+            JSON.stringify(serviceOptions.clientOptions),
```

Comment:
Is JSON.stringify guaranteed to return a stable ordering of keys for the same input? I think we have some "hash" code, or some other way of dealing with that, somewhere...